sdk/core/System.TypeSpec/src/PipelineOptions.cs

```diff
@@ -0,0 +1,99 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.ComponentModel;
+using System.Diagnostics;
+using System.ServiceModel.Rest.Core;
+using System.Threading;
+
+namespace System.ServiceModel.Rest;
+
+// Make options freezable
+public class PipelineOptions
+{
+    private object _lock = new object();
+    private MessagePipeline? _pipeline;
+    private volatile bool _isFrozen;
+
+    private IPipelinePolicy<PipelineMessage>? _retryPolicy;
+    private IPipelinePolicy<PipelineMessage>? _loggingPolicy;
+    private PipelineTransport<PipelineMessage> _transport;
+    private ReadOnlyMemory<IPipelinePolicy<PipelineMessage>> _perTryPolicies;
+    private ReadOnlyMemory<IPipelinePolicy<PipelineMessage>> _perCallPolicies;
+
+    public PipelineOptions(PipelineTransport<PipelineMessage> transport)
+    {
+        _transport = transport;
+    }
+
+    [EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
It looks like we call this from a lot of places in the demo client, e.g. the constructor, the protocol method, the method creation private helper method. Why would we EBN it?

Reply:
I wanted the public API to be only the user facing options.

Reply:
Interesting.  I'm wondering if we should split the types into user-facing and author-facing so we don't have to use EBN.  Let me think.